include/nuttx/mtd/mtd.h

```diff
@@ -74,6 +74,12 @@
 #define MTDIOC_ERASESTATE _MTDIOC(0x000a) /* IN:  Pointer to uint8_t
                                            * OUT: Byte value that represents the
                                            *      erased state of the MTD cell */
+#define MTDIOC_PARTINFO   _MTDIOC(0x000b) /* IN:  Pointer to write-able struct
```

Comment:
change to FIOC_PARTINFO?

Reply:
For me it does not seem fit to add it to the File-system IOCTL command group, since we're dealing with partitions here.
In my mind there were two possibilities:
1) Having both `MTDIOC_PARTINFO` and `BIOC_PARTINFO` which would behave similar to what we currently have for `MTDIOC_XIPBASE` and `BIOC_XIPBASE`.
2) Creating generic commands like Linux does, e.g. ` MEMGETPARTINFO`, which would be handled both by block devices and MTD.

I decided to stick to the first option because the second approach would require some refactoring on NuttX interfaces, which would cause some trouble to existing applications.

Reply:
Should we add BIOC_PARTINFO in this PR? I would prefer the userspace program use BIOC_PARTINFO instead MTDIOC_PARTINFO since this make the program work with both block and mtd device.

Reply:
Ok, I think we may already create the command definition in this PR.
I'll add it to the list of `ioctl` commands.

Reply:
Done. I've also made FTL translate an incoming `BIOC_PARTINFO` into `MTDIOC_PARTINFO`.